lib/src/iam/verify.rs

```diff
@@ -333,7 +370,24 @@ pub async fn token(kvs: &Datastore, session: &mut Session, token: &str) -> Resul
 			let mut tx = kvs.transaction(Read, Optimistic).await?;
 			// Get the namespace token
 			let de = tx.get_ns_token(&ns, &tk).await?;
-			let cf = config(de.kind, de.code)?;
+			// If the token is defined as JWKS
+			let cf =
+				if de.kind == Algorithm::Jwks {
+					#[cfg(not(feature = "jwks"))]
+					{
+						warn!("Failed to verify a token defined as JWKS when the feature is not enabled");
+						Err(Error::InvalidAuth)
+					}
+					#[cfg(feature = "jwks")]
+					// The key identifier header must be present
+					if let Some(kid) = token_data.header.kid {
+						jwks::config(kvs, &kid, &de.code).await
+					} else {
+						Err(Error::MissingTokenHeader("kid".to_string()))
+					}
+				} else {
+					config(de.kind, de.code)
+				}?;
```

Comment:
If I read it correctly, this block is repeated 2 more times above. Could we move it to a function to DRY it?

Reply:
I am not completely sure it is clearer this way, but I have moved it to its own function in [cf58797](https://github.com/surrealdb/surrealdb/pull/3198/commits/cf58797f2566c6e007632137291f9ce8e4d075c5).